plutus-contract/src/Plutus/Trace/Emulator.hs

```diff
@@ -82,26 +84,31 @@ import           Plutus.Trace.Effects.EmulatedWalletAPI  (EmulatedWalletAPI, han
 import qualified Plutus.Trace.Effects.EmulatedWalletAPI  as EmulatedWalletAPI
 import           Plutus.Trace.Effects.EmulatorControl    (EmulatorControl, handleEmulatorControl)
 import qualified Plutus.Trace.Effects.EmulatorControl    as EmulatorControl
-import           Plutus.Trace.Effects.RunContract        (RunContract, handleRunContract)
+import           Plutus.Trace.Effects.RunContract        (RunContract, handleRunContract, mapYieldEm)
 import qualified Plutus.Trace.Effects.RunContract        as RunContract
 import           Plutus.Trace.Effects.Waiting            (Waiting, handleWaiting)
 import qualified Plutus.Trace.Effects.Waiting            as Waiting
-import           Plutus.Trace.Emulator.ContractInstance  (EmulatorRuntimeError)
+import           Plutus.Trace.Emulator.ContractInstance  (EmulatorRuntimeError, handleContractRuntime)
 import           Plutus.Trace.Emulator.System            (launchSystemThreads)
 import           Plutus.Trace.Emulator.Types             (ContractConstraints, ContractHandle (..), ContractInstanceTag,
                                                           Emulator, EmulatorMessage (..), EmulatorThreads,
                                                           UserThreadMsg (..))
 import           Streaming                               (Stream)
 import           Streaming.Prelude                       (Of)
 
-type EmulatorTrace a =
-        Eff
-            '[ RunContract
-            , Waiting
-            , EmulatorControl
-            , EmulatedWalletAPI
-            , LogMsg String
-            ] a
+-- | Effects that make up an emulator trace:
+--   Running contracts, waiting for blockchain events,
+--   changing emultor internal state, and logging.
+type EmulatorTraceEffs effs =
```

Comment:
Not a major concern but for the sake of consistency it would be nice to agree on some syntax to use..

```haskell
type SomeEffs m = '[ Foo, Bar, Baz, m ]
```
vs

```haskell
type SomeEffs m = Foo ': Bar ': Baz :' m
```

vs

using Member/Members (but i understand that has some limitations so we can't really use that everywhere)

Reply:
`type SomeEffs m = '[ Foo, Bar, Baz, m ]` only allows you to choose the last element in the list, whereas `Foo ': Bar ': Baz :' m` lets you choose the entire rest of the list. I use `type SomeEffs effs = Foo ': Bar ': effs` when I want to have `'[Foo, Bar]` in some places (with `SomeEffs '[]`)  and some specific list of types in other places. 

But yes, more consistency would be nice.